src/MongoDB.Driver/MongoClient.cs

```diff
@@ -223,13 +231,37 @@ public sealed override IMongoDatabase GetDatabase(string name, MongoDatabaseSett
             return CreateDatabaseNamesCursor(databases);
         }
 
+        /// <inheritdoc />
```

Comment:
Change the previous overload to not do any work. Overloads that exist only to provide default values for optional parameters should delegate to another overload passing the default value:

    public sealed override IAsyncCursor<string> ListDatabaseNames(
        IClientSessionHandle session,
        CancellationToken cancellationToken = default(CancellationToken))
    {
        return ListDatabaseNames(session, options: null, cancellationToken);
    }

Same for corresponding async method.

Reply:
Done.